karapace/karapace.py

```diff
@@ -7,28 +7,19 @@
 
 from functools import partial
 from http import HTTPStatus
+from karapace.config import Config
 from karapace.rapu import HTTPResponse, RestApp
 from typing import NoReturn, Union
 
 import asyncio
 import logging
-import os
 
 
 class KarapaceBase(RestApp):
-    def __init__(self, config: dict) -> None:
-        sentry_config = config.get("sentry", {"dsn": None}).copy()
-        super().__init__(app_name="karapace", sentry_config=sentry_config)
+    def __init__(self, config: Config) -> None:
+        super().__init__(app_name="karapace", config=config)
 
         self.kafka_timeout = 10
-        self.config = config
-        self._sentry_config = sentry_config
-        if os.environ.get("SENTRY_DSN"):
-            self._sentry_config["dsn"] = os.environ["SENTRY_DSN"]
-        if "tags" not in self._sentry_config:
-            self._sentry_config["tags"] = {}
-        self._sentry_config["tags"]["app"] = "Karapace"
```

Comment:
This was super weird, the variable `self._sentry_config` is never used, but it refers to the same object as `sentry_config`, which is forwarded to `RestApp` then `StatsClient`, where these side effects are used.

For the sake of simplicity I changed this to be set with the other configs.